sys/kern/interrupt.c

```diff
@@ -157,30 +180,44 @@ void intr_event_run_handlers(intr_event_t *ie) {
 }
 
 static void intr_thread(void *arg) {
+  intr_thread_t *it = (intr_thread_t *)arg;
+
   while (true) {
-    intr_handler_t *ih;
+    intr_handler_t *ih = NULL, *next, *curr;
```

Comment:
I think that mixing declarations with assignment and without assignment is not a good idea.